compiler/AST/primitive.cpp

```diff
@@ -659,11 +659,21 @@ initPrimitive() {
   // dst, type to default-init
   prim_def(PRIM_DEFAULT_INIT_VAR, "default init var", returnInfoVoid);
 
+  // indicates split initialization point of declaration
+  // If the type is provided, apply that to the value immediately
+  // The value may not be used until a later PRIM_INIT_VAR.
+  //
+  // dst, optional type to default-init
+  prim_def(PRIM_INIT_VAR_SPLIT_DECL, "init var split decl", returnInfoVoid, false);
+
```

Comment:
I'd love to see PRIM_INIT_VAR_SPLIT_DECL placed between PRIM_INIT_VAR and PRIM_INIT_VAR_SPLIT_INIT.